src/numerics/basis/basis.py

```diff
@@ -563,6 +563,33 @@ def get_quadrature_data(self, order):
 
 		return quad_pts, quad_wts # [nq, ndims] and [nq, 1]
 
+	def get_tiling_constants(self, bface_quad_pts_st):
+		'''
+		Precomputes the tiling constants for the ADER-DG scheme. Tiling
+		constants are used with the numpy 'tile' function to build arrays
+		to maintain consistency for the tensor multiplications throughout
+		the solver.
+
+		Inputs:
+		-------
+			bface_quad_pts_st: boundary face quad_pts used to define
+					the time_skip and time_tile value for 2D ADER 
+					approaches using Tris [nq_st, ndims]
+
+		Outputs:
+		--------
+			nq_t: quadrature points tiling constant
+			time_skip: Value to skip when building time
+					array for each bface in get_boundary_face_residual
+					in src/solver/ADERDG.py. Also used for tiling
+					in the interior and boundary face integral.
+			time_tile: time array tiling constant for each bface in 
+					get_boundary_face_residual in src/solver/ADERDG.py
+		'''
+		return int(np.sqrt(bface_quad_pts_st.shape[0])), \
+				int(np.sqrt(bface_quad_pts_st.shape[0])), \
+				int(np.sqrt(bface_quad_pts_st.shape[0]))
+
```

Comment:
Is this the best place for this? Looks like an ADER-specific thing, maybe it should just be moved to the ADER volume helpers or something, not sure? Seems out of place in TriShape or ShapeBase

Reply:
Agreed. The tiling constants have been moved to the ader_helpers object in ADERDG. It checks the shape and then calls the corresponding get_tiling_constants_<Shape_Type> functions. This was added in cf4e98b446df69b3a5734d28999f10f7eddc793f